src/App.tsx

```diff
@@ -15,36 +17,83 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  String = 'alphabetical',
+  Number = 'length',
+  Default = '',
+}
+
 export const App: React.FC = () => {
+  const [sortOrder, setSortOrder] = useState<SortType>(SortType.Default);
+  const [reversed, setReversed] = useState<boolean>(false);
+  const sortedGoods = [...goodsFromServer].sort((good1: string, good2: string) => {
+    switch (sortOrder) {
+      case 'alphabetical':
+        return good1.localeCompare(good2);
+      case 'length':
+        return good1.length - good2.length;
```

Comment:
[CODE STYLE] The function name 'length' does not start with a verb. Consider renaming it to 'sortByLength' for better readability and to follow the checklist. It's a minor issue, but it's good for code readability.